src/riscv-integration.adoc

```diff
@@ -973,11 +976,29 @@ is the cause of the fault. The possible CHERI types and causes are encoded as
 shown in xref:mtval-cheri-type[xrefstyle=short] and
 xref:mtval-cheri-causes[xrefstyle=short] respectively.
 
+[#stid,reftext="stid"]
+==== Supervisor Thread Identifier (stid)
+
+The <<stid>> register is an SXLEN-bit read-write register. It is used to
+identify the current thread.
+
+.Supervisor thread identifier register
+include::img/stidreg.edn[]
+
 === Unprivileged CSRs
 
 Unlike machine and supervisor level CSRs, {cheri_base_ext_name} does not require
 <<pcc>> to grant <<asr_perm>> to access unprivileged CSRs.
 
+[#utid,reftext="utid"]
+==== User Thread Identifier (utid)
+
+The <<utid>> register is an UXLEN-bit read-write register. It is a read-only
```

Comment:
```suggestion
The <<utid>> register is an UXLEN-bit read-only register. It is a read-only
```
Do we really want these to be the same or have a separate user one that can only be written by S-mode?

Reply:
That would indicate that we commonly expect that there is not a 1-1 mapping of user threads to kernel threads. Once we have user space threading, the user space thread scheduler can provide a solution, e.g., giving out a read-only capability that contains the current user space thread ID.